src/differential_arithmetic.jl

```diff
@@ -67,11 +67,15 @@ for T in (:One, :AbstractThunk, :Any)
 end
 
 
-Base.:+(a::One, b::One) = extern(a) + extern(b)
+strong_one(primal::Number) = true
+strong_one(primal::AbstractMatrix) = LinearAlgebra.I
+strong_one(primal) = one(primal)
+
+Base.:+(a::One, b::One) = extern(a) + extern(b)  # This is not well-defined.
```

Comment:
i'd like this comment to be more informative, if possible (i.e "so why do we have this?!")

Reply:
Its not changed by this PR.

Reply:
It really should be discussed somewhere else, 
Maybe #56 

Descibing why this is a problem in the code,
is splitting information that belongs in issues.
Maybe i should just revert the change to this line.

```suggestion
Base.:+(a::One, b::One) = extern(a) + extern(b)
```